### PR TITLE
Fix missing logs issue

### PR DESCRIPTION
**Pull request by vishweshbankwar**
#34259 reported that logs are not exported if any one of the logs in the received batch hit an exception when we convert the log to Azure Monitor trace. This PR addresses this by handling the exception individually for each log rather than failing the entire batch.

### CODE REVIEW

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/AzureMonitorLogExporter.cs [hunk 1]
```diff
@@ -44,19 +44,26 @@ public override ExportResult Export(in Batch<LogRecord> batch)
             // Prevent Azure Monitor's HTTP operations from being instrumented.
             using var scope = SuppressInstrumentationScope.Begin();
 
+            ExportResult exportResult = ExportResult.Failure;
+
             try
             {
                 var telemetryItems = LogsHelper.OtelToAzureMonitorLogs(batch, LogResource, _instrumentationKey);
-                var exportResult = _transmitter.TrackAsync(telemetryItems, false, CancellationToken.None).EnsureCompleted();
+                if (telemetryItems.Count > 0)
```
Comment (TimothyMothra) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/AzureMonitorLogExporter.cs:52]: Do we need a similar change in the other Exporters?

**Review by rajkumar-rangaraj — APPROVED** [review]

**Review by TimothyMothra — APPROVED** [review]

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/AzureMonitorLogExporter.cs (hunk 1 above)
Comment (vishweshbankwar) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/AzureMonitorLogExporter.cs:52]: Yes, was planning in separate one but just pushed the changes here.

**Comment by reyang** [comment]
@cijothomas FYI you've mentioned that #34259 might depend on a fix in the OpenTelemetry .NET SDK, @vishweshbankwar do we know if this PR is going to fix #34259 or we need other changes?

**Comment by vishweshbankwar** [comment]
> @cijothomas FYI you've mentioned that #34259 might depend on a fix in the OpenTelemetry .NET SDK, @vishweshbankwar do we know if this PR is going to fix #34259 or we need other changes?

We need other changes. There are two parts that need a fix
1) Missing logs (This is a side effect of error shown in the issue) --> Fixed by this PR
2) Error shown in the issue --> This needs further investigation.

```csharp
Object name: 'Collection'.
   at Microsoft.AspNetCore.Http.Features.FeatureReferences`1.ThrowContextDisposed()
   at Microsoft.AspNetCore.Http.Features.FeatureReferences`1.ContextDisposed()
   at Microsoft.AspNetCore.Http.Features.FeatureReferences`1.Fetch[TFeature](TFeature& cached, Func`2 factory)
   at Microsoft.AspNetCore.Http.DefaultHttpRequest.get_Protocol()
   at Microsoft.AspNetCore.Hosting.HostingRequestStartingLog.get_Item(Int32 index)
   at Microsoft.AspNetCore.Hosting.HostingRequestStartingLog.GetEnumerator()+MoveNext()
   at Azure.Monitor.OpenTelemetry.Exporter.Internals.LogsHelper.ExtractProperties(String& message, IDictionary`2 properties, IReadOnlyCollection`1 stateDictionary)
   at Azure.Monitor.OpenTelemetry.Exporter.Internals.LogsHelper.GetMessageAndSetProperties(LogRecord logRecord, IDictionary`2 properties)
   at Azure.Monitor.OpenTelemetry.Exporter.Models.MessageData..ctor(Int32 version, LogRecord logRecord)
   at Azure.Monitor.OpenTelemetry.Exporter.Internals.LogsHelper.OtelToAzureMonitorLogs(Batch`1 batchLogRecord, AzureMonitorResource resource, String instrumentationKey)
   at Azure.Monitor.OpenTelemetry.Exporter.AzureMonitorLogExporter.Export(Batch`1& batch)]
```

**Comment by reyang** [comment]
> We need other changes. There are two parts that need a fix

Got it, would you reopen the issue + update the PR description here to reflect the intention? Thanks!

**Comment by vishweshbankwar** [comment]
> > We need other changes. There are two parts that need a fix
> 
> Got it, would you reopen the issue + update the PR description here to reflect the intention? Thanks!

@reyang I have updated the description here and sent another PR to handle part 2 i.e. the exception in a separate PR.

**Pull request merged**